frontend/src/reducers/reportReducer.js

```diff
@@ -12,11 +15,38 @@ const reportReducer = (state = initialState, action) => {
         ...state,
         mineReports: action.payload.records,
       };
+    case actionTypes.STORE_MINE_REPORT_COMMENTS:
+      return {
+        ...state,
+        reportComments: action.payload.records,
+        commentsLoading: false,
+      };
+    case actionTypes.FETCH_MINE_REPORT_COMMENTS:
+      return {
+        ...state,
+        commentsLoading: true,
+      };
+    case actionTypes.SUBMIT_MINE_REPORT_COMMENT:
+      return {
+        ...state,
+        commentSubmitting: true,
+      };
+    case actionTypes.MINE_REPORT_COMMENT_CREATED:
+      return {
+        ...state,
+        commentSubmitting: false,
```

Comment:
I don't have a lot of context on what you're up to with comments, but from the outside looking in, these look like component-level state concerns that are being handled in reducers. Separation of concerns issue?

Reply:
In this case I find it hard to disagree with you. I think there could be a case for bringing state concerns into the flux store (ongoing jobs, jobs with dependencies on other components) but not in this case.